donut/modules/bodfeedback/routes.py

```diff
@@ -0,0 +1,168 @@
+        return
+    helpers.add_msg(complaint_id, data['message'], data['poster'])
+    return flask.redirect(
+        flask.url_for('bodfeedback.bodfeedback_view_complaint', id=id))
+
+
+# Allow bod members to see a summary
+@blueprint.route('/bodfeedback/view/summary')
+def bodfeedback_view_summary():
+    if 'username' not in flask.session or not auth_utils.check_permission(
+            flask.session['username'], bod_permissions.SUMMARY):
+        flask.abort(403)
+        return
+    # Get a list containing data for each post
+    complaints = helpers.get_new_posts()
+    # Add links to each complaint
+    for complaint in complaints:
+        complaint['link'] = helpers.get_link(complaint['complaint_id'])
+    return flask.render_template('summary.html', complaints=complaints)
+
+
+# Mark a complaint read
+@blueprint.route('/1/bodfeedback/markRead/<uuid:id>')
+def bodfeedback_mark_read(id):
+    # Authenticate
+    if 'username' not in flask.session or not auth_utils.check_permission(
+            flask.session['username'], bod_permissions.TOGGLE_READ):
+        flask.abort(403)
+        return
+    complaint_id = helpers.get_id(id)
+    if helpers.mark_read(complaint_id) == False:
+        flask.abort(400)
+        return
+    return 'Success'
+
+
+# Mark a complaint unread
+@blueprint.route('/1/bodfeedback/markUnread/<uuid:id>')
+def bodfeedback_mark_unread(id):
+    # Authenticate
+    if 'username' not in flask.session or not auth_utils.check_permission(
+            flask.session['username'], bod_permissions.TOGGLE_READ):
+        flask.abort(403)
+        return
+    complaint_id = helpers.get_id(id)
+    if helpers.mark_unread(complaint_id) == False:
+        flask.abort(400)
+        return
+    return 'Success'
+
+
+# Add an email to this complaint
+@blueprint.route('/1/bodfeedback/addEmail/<uuid:id>', methods=['POST'])
+def bodfeedback_add_email(id):
+    if 'username' not in flask.session or not auth_utils.check_permission(
+            flask.session['username'], bod_permissions.ADD_REMOVE_EMAIL):
+        flask.abort(403)
+        return
+    complaint_id = helpers.get_id(id)
+    if not complaint_id:
+        flask.abort(400)
+        return
+    data = {}
+    data['email'] = flask.request.form.get('email')
+    if data['email']:
+        helpers.add_email(complaint_id, data['email'])
+    return flask.redirect(
+        flask.url_for('bodfeedback.bodfeedback_view_complaint', id=id))
+
+
+# Remove an email from this complaint
+@blueprint.route('/1/bodfeedback/removeEmail/<uuid:id>', methods=['POST'])
+def bodfeedback_remove_email(id):
+    if 'username' not in flask.session or not auth_utils.check_permission(
+            flask.session['username'], bod_permissions.ADD_REMOVE_EMAIL):
+        flask.abort(403)
+        return
+    complaint_id = helpers.get_id(id)
+    if not complaint_id:
+        flask.abort(400)
+        return
+    data = flask.request.form.getlist('emails')
+    if data:
+        for em in data:
```

Comment:
If `data` is empty, the `for` loop should never run anyways, so I think `if data` is redundant